lib/Routing/RedirectHandler.php

```diff
@@ -174,25 +174,41 @@ protected function buildRedirectResponse(Redirect $redirect, Request $request, $
             $url = replace_pcre_backreferences($url, $matches);
         }
 
-        if ($redirect->getTargetSite() && !preg_match('@http(s)?://@i', $url)) {
-            if ($targetSite = Site::getById($redirect->getTargetSite())) {
-                // if the target site is specified and and the target-path is starting at root (not absolute to site)
-                // the root-path will be replaced so that the page can be shown
-                $url = preg_replace('@^' . $targetSite->getRootPath() . '/@', '/', $url);
-                $url = $request->getScheme() . '://' . $targetSite->getMainDomain() . $url;
+        if(!preg_match('@http(s)?://@i', $url)) {
+            if ($redirect->getTargetSite()) {
+                if ($targetSite = Site::getById($redirect->getTargetSite())) {
+                    // if the target site is specified and and the target-path is starting at root (not absolute to site)
+                    // the root-path will be replaced so that the page can be shown
+                    $url = preg_replace('@^' . $targetSite->getRootPath() . '/@', '/', $url);
+                    $url = $request->getScheme() . '://' . $targetSite->getMainDomain() . $url;
+                } else {
+                    $this->logger->error('Site with ID {targetSite} not found', [
+                        'redirect' => $redirect->getId(),
+                        'targetSite' => $redirect->getTargetSite()
+                    ]);
+
+                    return null;
+                }
             } else {
-                $this->logger->error('Site with ID {targetSite} not found', [
-                    'redirect' => $redirect->getId(),
-                    'targetSite' => $redirect->getTargetSite()
-                ]);
+                $redirectDomain = $this->config['general']['domain'];
+                $sites = new Site\Listing();
+                foreach ($sites as $site) {
+                    foreach ($site->getDomains() as $siteDomain) {
+                        if ($request->getHost() === $siteDomain) {
```

Comment:
why not consider site main domain also for comparison? and better use something like array_search() instead of second foreach

Reply:
Done. I use `array_searchi()` as domain names are not case-sensitive.